File: Frontend/petture/src/app/models/commentary.ts
export class Commentary {

    constructor(
        private _id: number,
        private _userName: string,
        private _body: string,
        private _postId: number
    ){}

    public get body(): string {
        return this._body;
    }
    public set body(value: string) {
        this._body = value;
    }
    public get userName(): string {
        return this._userName;
    }
    public set userName(value: string) {
        this._userName = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get postId(): number {
        return this._postId;
    }
    public set postId(value: number) {
        this._postId = value;
    }
}

File: Backend/post-service/src/main/java/com/ironhack/postservice/model/Commentary.java
package com.ironhack.postservice.model;

import com.ironhack.postservice.utils.dtos.*;

import javax.persistence.*;

@Entity
public class Commentary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String userName;
    private String body;
    @ManyToOne
    @JoinColumn(name = "post_id")
    private Post post;

//    Constructors

    public Commentary() {
    }

    public Commentary(String userName, String body, Post post) {
        this.userName = userName;
        this.body = body;
        this.post = post;
    }

//    Getters and Setters

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }
}
